pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyService.java

```diff
@@ -347,7 +349,9 @@ public void close() throws IOException {
             proxyAdditionalServlets = null;
         }
 
-        metricsServlet = null;
+        synchronized (this) {
```

Comment:
what about using a synchronised method here instead of using "synchronized (this)" ?
probably it is a matter of taste, but I think that "synchronized (this)" is quite hard to read